mongoOperator/helpers/RestoreHelper.py

```diff
@@ -0,0 +1,174 @@
+        """
+        Retrieves the storage credentials for the given cluster object from the Kubernetes secret as specified in the
+        cluster object.
+        :param cluster_object: The cluster object from the YAML file.
+        :return: The credentials dictionary.
+        """
+        secret_key = cluster_object.spec.backups.gcs.service_account.secret_key_ref
+        secret = self.kubernetes_service.getSecret(secret_key.name, cluster_object.metadata.namespace)
+        credentials_encoded = secret.data[secret_key.key]
+        credentials_json = b64decode(credentials_encoded)
+        return json.loads(credentials_json)
+
+    def getLastBackup(self, cluster_object: V1MongoClusterConfiguration) -> str:
+        """
+        Returns the filename of the last backup file in the bucket.
+        :param cluster_object: The cluster object from the YAML file.
+        :return: String containing the filename of the last backup.
+        """
+        prefix = cluster_object.spec.backups.gcs.prefix or self.DEFAULT_BACKUP_PREFIX
+        bucket_name = cluster_object.spec.backups.gcs.restore_bucket if cluster_object.spec.backups.gcs.restore_bucket \
+            else cluster_object.spec.backups.gcs.bucket
+        return self._lastBackupFile(
+            credentials=self._getCredentials(cluster_object),
+            bucket_name=bucket_name,
+            key="{}/".format(prefix)
+        )
+
+    @staticmethod
+    def _lastBackupFile(credentials: dict, bucket_name: str, key: str) -> str:
+        """
+        Gets the name of the last backup file in the bucket.
+        :param credentials: The Google cloud storage service credentials retrieved from the Kubernetes secret.
+        :param bucket_name: The name of the bucket.
+        :param key: The prefix of tha backups
+        :return: The location of the last backup file.
+        """
+        credentials = ServiceCredentials.from_service_account_info(credentials)
+        gcs_client = StorageClient(credentials.project_id, credentials)
+        bucket = gcs_client.get_bucket(bucket_name)
+        blobs = bucket.list_blobs(prefix=key)
+
+        last_blob = None
+        for blob in blobs:
+            logging.info("Found backup file '%s' in bucket '%s'", blob.name, bucket_name)
+            if last_blob is None or blob.time_created > last_blob.time_created:
+                last_blob = blob
+
+        logging.info("Returning backup file %s", last_blob.name.replace(key, ""))
+        return last_blob.name.replace(key, "") if last_blob else None
+
+    def restoreIfNeeded(self, cluster_object: V1MongoClusterConfiguration) -> bool:
+        """
+        Checks whether a restore is requested for the cluster, looking up the restore file if
+        necessary.
+        :param cluster_object: The cluster object from the YAML file.
+        :return: Whether a restore was executed or not.
+        """
+        if cluster_object.spec.backups.gcs.restore_from is not None:
```

Comment:
We should use early return here to prevent nested code complexity:

```python
if cluster_object.spec.backups.gcs.restore_from is None:
    return False

backup_file = cluster_object.spec.backups.gcs.restore_from
if backup_file == "latest":
    backup_file = self.getLastBackup(cluster_object)

logging.info(...)

return True
```

This whey there is only 1 level of complexity in 1 lines.

Reply:
Shouldn't there be a default value for restore_from? As it's not required in the model.